src/base/filesystemwatcher.cpp

```diff
@@ -1,43 +1,68 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "filesystemwatcher.h"
+
 #include <QtGlobal>
+
 #ifndef Q_OS_WIN
-#include <QSet>
+#include <cerrno>
 #include <iostream>
-#include <errno.h>
+#include <QSet>
+#endif
+
 #if defined(Q_OS_MAC) || defined(Q_OS_FREEBSD)
-#include <sys/param.h>
+#include <cstring>
 #include <sys/mount.h>
-#include <string.h>
-#elif !defined Q_OS_HAIKU
+#include <sys/param.h>
+#elif !defined Q_OS_WIN && !defined Q_OS_HAIKU
 #include <sys/vfs.h>
 #endif
-#endif
 
 #include "algorithm.h"
 #include "base/bittorrent/magneturi.h"
 #include "base/bittorrent/torrentinfo.h"
 #include "base/preferences.h"
-#include "filesystemwatcher.h"
 
-#ifndef CIFS_MAGIC_NUMBER
-#define CIFS_MAGIC_NUMBER 0xFF534D42
-#endif
-
-#ifndef NFS_SUPER_MAGIC
-#define NFS_SUPER_MAGIC 0x6969
-#endif
-
-#ifndef SMB_SUPER_MAGIC
-#define SMB_SUPER_MAGIC 0x517B
-#endif
+namespace
+{
+    const unsigned long CIFS_MAGIC_NUMBER = 0xFF534D42;
```

Comment:
I don't think it is necessary because it is usually defined in linux kernel headers, from http://man7.org/linux/man-pages/man2/statfs.2.html:
> Most of these MAGIC constants are defined in /usr/include/linux/magic.h, and some are hardcoded in kernel sources.

and our CI passes.

Or you insist adding #if conditionals?

Reply:
>Or you insist adding #if conditionals?

No.